In [16]:
!pip install -U pandasql

In [11]:
import pandas as pd

In [12]:
import sqlite3

In [9]:
!pip install ipython-sql

In [13]:
conn = sqlite3.connect('creditscoringmodel')

In [17]:
%load_ext sql

In [18]:
%sql sqlite:///creditscoringmodel

In [19]:
%%sql
CREATE TABLE CreditScoringModel (
    Factor VARCHAR(50),
    Weight_Percentage DECIMAL(5, 2),
    Average_Score_Impact DECIMAL(5, 2),
    Standard_Deviation DECIMAL(5, 2),
    Median_Impact DECIMAL(5, 2),
    Max_Impact DECIMAL(5, 2),
    Min_Impact DECIMAL(5, 2)
);

 * sqlite:///creditscoringmodel
Done.


[]

In [20]:
%%sql
INSERT INTO CreditScoringModel (Factor, Weight_Percentage, Average_Score_Impact, Standard_Deviation, Median_Impact, Max_Impact, Min_Impact)
VALUES
('Payment History', 35.00, 297.00, 50.00, 300.00, 350.00, 200.00),
('Credit Utilization', 30.00, 255.00, 40.00, 250.00, 300.00, 150.00),
('Length of Credit History', 15.00, 127.50, 20.00, 125.00, 150.00, 80.00),
('Types of Credit Used', 10.00, 85.00, 10.00, 90.00, 100.00, 50.00),
('Recent Credit Inquiries', 10.00, 85.00, 15.00, 80.00, 100.00, 50.00);

 * sqlite:///creditscoringmodel
5 rows affected.


[]

In [21]:
%%sql
SELECT * FROM CreditScoringModel;

 * sqlite:///creditscoringmodel
Done.


[('Payment History', 35, 297, 50, 300, 350, 200),
 ('Credit Utilization', 30, 255, 40, 250, 300, 150),
 ('Length of Credit History', 15, 127.5, 20, 125, 150, 80),
 ('Types of Credit Used', 10, 85, 10, 90, 100, 50),
 ('Recent Credit Inquiries', 10, 85, 15, 80, 100, 50)]

### Provides a comprehensive view of credit scoring factors and their impacts.


## Factor with the Highest Weight

In [22]:
%%sql
SELECT Factor
FROM CreditScoringModel
ORDER BY Weight_Percentage DESC
LIMIT 1;

 * sqlite:///creditscoringmodel
Done.


[('Payment History',)]

### "Payment History" is the most influential factor with a 35% weight, emphasizing its critical role in credit scoring.

## Average Standard Deviation

In [23]:
%%sql
SELECT AVG(Standard_Deviation) AS Average_Standard_Deviation
FROM CreditScoringModel;

 * sqlite:///creditscoringmodel
Done.


[(27.0,)]

### Calculates variability in score impacts, indicating how consistent factors affect credit scores.

## Factors with an Average Score Impact Above 200 Points

In [25]:
%%sql
SELECT Factor
FROM CreditScoringModel
WHERE Average_Score_Impact > 200;

 * sqlite:///creditscoringmodel
Done.


[('Payment History',), ('Credit Utilization',)]

### Highlights "Payment History" and "Credit Utilization" as significant factors with impacts above 200 points, guiding focus areas for score improvement.

## Range of Impacts for Each Factor

In [26]:
%%sql
SELECT Factor, (Max_Impact - Min_Impact) AS Impact_Range
FROM CreditScoringModel;

 * sqlite:///creditscoringmodel
Done.


[('Payment History', 150),
 ('Credit Utilization', 150),
 ('Length of Credit History', 70),
 ('Types of Credit Used', 50),
 ('Recent Credit Inquiries', 50)]

### Shows variability in factor impacts; "Payment History" has the widest range (150 points), suggesting it can strongly influence score fluctuations.

# Conclusion

## These analyses provide insights into the structure and influence of the credit scoring model used by US banks. Understanding these factors helps in strategizing to improve creditworthiness and manage risk effectively.